Add consensus articles (#1251)
iotaledger/iota-wiki: docs/learn/protocols/iota2.0/core-concepts/consensus/introduction.md

Under good network conditions, nodes collectively produce the same slot commitments. Under network partitioning, there could be a divergence in the perception of the suffix of the slot commitment chain. However, the finalized part of the slot commitment chain is safe under any network conditions. When the network converges, all honest nodes eventually switch to the heaviest slot commitment chain with the latest [finalized slot commitment](#finalization), thanks to the **[chain switching rule](chain-switching-rule.md)**.

## The Consensus Protocol’s Two Dimensions

The consensus protocol can be divided in two separate dimensions charged with different responsibilities:

1. [Agreement tasks for blocks and transactions](#agreement-tasks-for-blocks-and-transactions).
2. [Provide Consensus Flags](#provide-consensus-flags).

### Agreement Tasks for Blocks and Transactions

The first dimension consists of the agreement tasks for blocks and transactions. It has two main tasks:

> #### Broadcast blocks reliably

Reviewer: I would stick with a more standard terminology as it was in the original document "Reliable broadcast of blocks"

This task ensures the delivery of newly created blocks to all nodes in the network. Reliable broadcast guarantees that if an honest node includes a block in its locally maintained copy of the Tangle, all other honest nodes will eventually receive and deliver that block, including it in their locally maintained version.

#### Agreement on conflicting transactions

This task focuses on resolving conflicts that arise when multiple transactions in different blocks in the Tangle conflict with each other, attempting to consume the same output. By fulfilling this task, all honest nodes can reach a consensus on which conflicting transaction appears in the ledger, ensuring a consistent view across the network.

### Provide Consensus Flags

The second dimension provides different _consensus flags_ or confidence levels about the irreversibility of blocks and transactions.

When they receive a new validation block, nodes examine the Tangle to identify distinct _validation block patterns_, and attribute specific flags to blocks in the Tangle or transactions contained in the blocks.

> A pattern corresponds to a causally ordered set of validation blocks. For instance, if a block (object) is [approved](preliminaries.md#about-blocks-and-the-tangle) by a supermajority of the committee (validators holding more than $2/3$ of the total voting power), then the block gets the [pre-confirmed flag](#pre-confirmation). In this context, the validation blocks from these validators constitute a validation block pattern necessary for pre-confirmation.

Reviewer: ```suggestion
A pattern corresponds to a causally ordered set of validation blocks. For instance, if a block is [approved](preliminaries.md#about-blocks-and-the-tangle) by a supermajority of the committee (validators holding more than $2/3$ of the total voting power), then the block gets the [pre-confirmed flag](#pre-confirmation). In this context, the validation blocks from these validators constitute a validation block pattern necessary for pre-confirmation.
```

The IOTA 2.0 consensus protocol is endowed with dynamic availability. This attribute ensures continuous operation even when some validator committee members may be offline or the network experiences partitions. During such instances, the online validator committee within each partition maintains the process of updating their _available_ ledger, which is constructed from the transactions that have been [accepted](#acceptance).

Once the network recovers from its divided state and a [supermajority](preliminaries.md#total-and-online-supermajority) of the committee is back online, the transaction finalization process recommences. The consensus flags differentiate between degrees of irreversibility. This aspect balances safety and liveness, providing different trade-offs to manage various network conditions effectively.

:::info

> You can learn more about a formal description of consensus protocols supporting both safety and liveness in the [Ebb-and-FlowProtocols: AResolutionoftheAvailability-FinalityDilemma paper](https://arxiv.org/pdf/2009.04987).

Author: ```suggestion
You can learn more about a formal description of consensus protocols supporting both safety and liveness in the [Ebb and Flow Protocols: A Resolution of the Availability-Finality Dilemma  paper](https://arxiv.org/pdf/2009.04987).
```

#### Pre-Confirmation

The [Pre-Confirmation flag](consensus-flags.md#pre-confirmation-flag) is an internal flag used to define Confirmation. This flag is raised for a block when the block is causally referenced by a [supermajority of the total committee](preliminaries.md#total-and-online-supermajority), not just those currently online. It signifies a higher approval level than the pre-acceptance flag, but it is raised slowly under bad network conditions.

#### Confirmation

The [Confirmation flag](consensus-flags.md#confirmation-flag) is a flag that is exposed to the user in the wallet. For a block, it shows whether the block has been acknowledged by a [total supermajority](preliminaries.md#total-and-online-supermajority) of pre-confirmed validation blocks. This flag is raised for a transaction when it is accepted, and one of its attachments (blocks containing this transaction) gets confirmed. The confirmation flag favors safety and can be slow under bad network conditions.

#### Finalization

> The [Finalization](consensus-flags.md#finalization-flag) is a flag that is exposed to the user in the wallet. This block or transaction flag relies on finalizing the slot commitment that commits to the block or transaction. The latter means that a block containing a given [slot commitment](preliminaries.md#slot-commitment-chain) (and all objects assigned to it) gets [confirmed](consensus-flags.md#confirmation-flag). This flag is safe under any network conditions.

Reviewer: ```suggestion
The [Finalization flag](consensus-flags.md#finalization-flag) is a flag that is exposed to the user in the wallet. This block or transaction flag relies on finalizing the slot commitment that commits to the block or transaction. The latter means that a block containing a given [slot commitment](preliminaries.md#slot-commitment-chain) (and all objects assigned to it) gets [confirmed](consensus-flags.md#confirmation-flag). This flag is safe under any network conditions.
```
